include/deal.II/base/point.h

```diff
@@ -371,7 +371,8 @@ inline Point<dim, Number>::Point(const Number x, const Number y, const Number z)
 template <int dim, typename Number>
 inline Point<dim, Number>::Point(
   const boost::geometry::model::
-    point<Number, dim, boost::geometry::cs::cartesian> boost_pt)
+    point<Number, dim, boost::geometry::cs::cartesian> *boost_pt,
+  typename std::enable_if<dim != 0, int>::type = 0)
```

Comment:
The constructor itself does not have any template arguments and `dim` is at this point already resolved. Hence, this will not work. You need to template the constructor as well with a default template parameter equal to `dim` and then use that one instead of `dim` in `std::enable_if`. You might want to check via a `static_assert` in the body that the new template parameter and `dim` are equal indeed.